main/exercise/ptest_exercise_report.php

```diff
@@ -0,0 +1,504 @@
+            var defaultGroupId = Cookies.get('default_group_<?php echo $exercise_id; ?>');
+            $('#gs_group_name').val(defaultGroupId);
+            // Adding search options
+            var options = {
+                'stringResult': true,
+                'autosearch' : true,
+                'searchOnEnter': false,
+                afterSearch: function () {
+                    $('#gs_group_name').on('change', function() {
+                        var defaultGroupId = $('#gs_group_name').val();
+                        // Save default group id
+                        Cookies.set('default_group_<?php echo $exercise_id; ?>', defaultGroupId);
+                    });
+                }
+            }
+            jQuery("#results").jqGrid('filterToolbar', options);
+            sgrid.triggerToolbar();
+            $('#results').on('click', 'a.exercise-recalculate', function (e) {
+                e.preventDefault();
+                if (!$(this).data('user') || !$(this).data('exercise') || !$(this).data('id')) {
+                    return;
+                }
+                var url = '<?php echo api_get_path(WEB_CODE_PATH); ?>exercise/recalculate.php?<?php echo api_get_cidreq(); ?>';
+                var recalculateXhr = $.post(url, $(this).data());
+                $.when(recalculateXhr).done(function (response) {
+                    $('#results').trigger('reloadGrid');
+                });
+            });
+        <?php
+        }
+        ?>
+        });
+    // datepicker functions
+    var datapickerInputModified = false;
+    /**
+     * return true if the datepicker input has been modified
+     */
+    function datepicker_input_changed() {
+        datapickerInputModified = true;
+    }
+
+    /**
+    * disply the datepicker calendar on mouse over the input
+    */
+    function datepicker_input_mouseover() {
+        $('#datepicker_start').datepicker( "show" );
+    }
+
+    /**
+    * display or hide the datepicker input, calendar and button
+    */
+    function display_date_picker() {
+        if (!$('#datepicker_span').is(":visible")) {
+            $('#datepicker_span').show();
+            $('#datepicker_start').datepicker( "show" );
+        } else {
+            $('#datepicker_start').datepicker( "hide" );
+            $('#datepicker_span').hide();
+        }
+    }
+
+    /**
+    * confirm deletion
+    */
+    function submit_datepicker() {
+        if (datapickerInputModified) {
+            var dateTypeVar = $('#datepicker_start').datepicker('getDate');
+            var dateForBDD = $.datepicker.formatDate('yy-mm-dd', dateTypeVar);
+            // Format the date for confirm box
+            var dateFormat = $( "#datepicker_start" ).datepicker( "option", "dateFormat" );
+            var selectedDate = $.datepicker.formatDate(dateFormat, dateTypeVar);
+            if (confirm("<?php echo convert_double_quote_to_single(get_lang('AreYouSureDeleteTestResultBeforeDateD')).' '; ?>" + selectedDate)) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line exceeds maximum limit of 120 characters; contains 145 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5018478)